google/ads/google_ads/client.py

```diff
@@ -454,35 +418,31 @@ def _get_customer_id(self, request):
         """
         return getattr(request, 'customer_id', None)
 
-    def _parse_response_to_json(self, response, exception):
-        """Parses response object to JSON.
+    def _parse_exception_to_str(self, exception):
+        """Parses response exception object to str for logging.
 
         Returns:
-            A str of JSON representing a response or exception from the
-            service.
+            A str representing a exception from the API.
 
         Args:
-            response: A grpc.Call/grpc.Future instance.
             exception: A grpc.Call instance.
         """
-        if exception:
-            # try to retrieve the .failure property of a GoogleAdsFailure.
-            failure = getattr(exception, 'failure', None)
-
-            if failure:
-                return _parse_message_to_json(failure)
-            else:
+        try:
+            # If the exception is from the Google Ads API the failure attribute
+            # will be an instance of GoogleAdsFailure and can be concatenated
+            # into a log string.
+            return exception.failure
+        except AttributeError:
+            try:
                 # if exception.failure isn't present then it's likely this is a
-                # transport error with a .debug_error_string method.
-                try:
-                    debug_string = exception.debug_error_string()
-                    return _parse_to_json(json.loads(debug_string))
-                except (AttributeError, ValueError):
-                    # if both attempts to retrieve serializable error data fail
-                    # then simply return an empty JSON string
-                    return '{}'
-        else:
-            return _parse_message_to_json(response.result())
+                # transport error with a .debug_error_string method and the
+                # returned JSON string will need to be formatted.
+                debug_string = exception.debug_error_string()
```

Comment:
do you need a throwaway `debug_string` variable?

Reply:
Nope, I changed it.